include/onnxruntime/core/platform/threadpool.h

```diff
@@ -314,7 +314,7 @@ class ThreadPool {
     }
 
     if (num_batches <= 0) {
-      num_batches = std::min<ptrdiff_t>(total, DegreeOfParallelism(tp));
+      num_batches = std::min<std::ptrdiff_t>(total, DegreeOfParallelism(tp));
```

Comment:
>num_batches = std::min<std::ptrdiff_t>(total, DegreeOfParallelism(tp)); [](http://example.com/codeflow?start=6&length=71)

why is std:: needed here?

Reply:
I ran into a compilation error on the server in one of the build pipelines. In this file ptrdiff is referenced about 40 times, everywhere except in this line it is used with std scope. This was somehow didn't throw error until now. I had to fix it.

Reply:
The following is the error I get on dev4 it i don't specify the scope.
/bert_ort/sajandhy/onnxruntime/include/onnxruntime/core/platform/threadpool.h:317:30: error: ‘ptrdiff_t’ was not declared in this scope